### PR TITLE
[DOCS] Remove and redirect frozen index overview content

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Removes and adds redirects for the frozen indices [overview][0], [best practices][1], [search][2], and [monitoring][3] pages.
* Removes glossary terms related to frozen indices.
* Updates several xrefs to point to the freeze index API docs.

Relates to #72946 and #70192.

### Preview
Redirect preview: https://elasticsearch_72990.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/frozen-indices.html

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/7.12/frozen-indices.html
[1]: https://www.elastic.co/guide/en/elasticsearch/reference/master/best_practices.html
[2]: https://www.elastic.co/guide/en/elasticsearch/reference/master/searching_a_frozen_index.html
[3]: https://www.elastic.co/guide/en/elasticsearch/reference/master/monitoring_frozen_indices.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/ilm/ilm-overview.asciidoc [hunk 1]
```diff
@@ -18,8 +18,7 @@ include::../glossary.asciidoc[tag=rollover-def-short]
 include::../glossary.asciidoc[tag=shrink-def-short]
 * **Force merge**: 
 include::../glossary.asciidoc[tag=force-merge-def-short]
-* **Freeze**:
-include::../glossary.asciidoc[tag=freeze-def-short]
+* **Freeze**: Makes an index read-only and minimizes its memory footprint.
```
Comment (jrodewig) [inline docs/reference/ilm/ilm-overview.asciidoc:21]: This copy comes from the removed `freeze` glossary term.

**Review by danhermann — APPROVED** [review]
Overall, the changes LGTM. I do wonder if we should remove references to "minimizing memory usage" and "minimal cluster overhead" since it implies that frozen indices have a lighter footprint than regular indices when they are now, in fact, no lighter in weight than any other index.

**Comment by jrodewig** [comment]
Thanks @danhermann.

> I do wonder if we should remove references to "minimizing memory usage" and "minimal cluster overhead" since it implies that frozen indices have a lighter footprint than regular indices when they are now, in fact, no lighter in weight than any other index.

This is great feedback. I've push a few commits to remove those references.

**Pull request merged**